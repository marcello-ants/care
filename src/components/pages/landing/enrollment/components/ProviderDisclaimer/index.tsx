// External Dependencies
import { Typography, Link } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

export default function ProviderDisclaimer() {
  // Styles
  const classes = useStyles();

  return (
    <Typography careVariant="disclaimer1" className={classes.disclaimer}>
      <span>
        You must be 18 years or older to use Care.com. By clicking &quot;Submit,&quot; you agree to
        our
      </span>{' '}
      <Link href="/terms-of-use-p1012.html" careVariant="disclaimer1">
        Terms of Use
      </Link>{' '}
      <span>and</span>{' '}
      <Link href="/privacy-policy-p1013.html" careVariant="disclaimer1">
        Privacy Policy
      </Link>
      <span>.</span>
    </Typography>
  );
}
