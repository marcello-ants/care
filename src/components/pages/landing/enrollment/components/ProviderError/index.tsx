// External Dependencies
import { Box } from '@material-ui/core';
import { Icon24UtilityWarning } from '@care/react-icons';
import { Typography } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

function ProviderError() {
  // Styles
  const classes = useStyles();

  return (
    <Box className={classes.errorBanner}>
      <Box>
        <Icon24UtilityWarning color="#CF230C" />
      </Box>
      <Box>
        <Typography variant="h4">Whoops! Something went wrong</Typography>
        <Typography careVariant="body3">
          There was an error when we tried to submit your request. Please try again.
        </Typography>
      </Box>
    </Box>
  );
}

export default ProviderError;
