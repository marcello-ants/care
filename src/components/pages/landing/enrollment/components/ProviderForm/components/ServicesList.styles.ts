import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  empty: {
    '& > .MuiSelect-select': {
      color: theme.palette.care?.grey[700],
      fontWeight: 'normal',
    },
  },
  inputContainer: {
    width: '100%',
    '& > div': {
      padding: 0,
    },
  },
  error: {
    color: theme?.palette?.care?.red[800],
    marginLeft: theme.spacing(1.25),
    marginTop: theme.spacing(0.5),
  },
}));

export default useStyles;
