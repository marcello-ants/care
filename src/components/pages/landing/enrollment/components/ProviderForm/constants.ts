export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
