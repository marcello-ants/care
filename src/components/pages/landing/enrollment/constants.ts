export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
