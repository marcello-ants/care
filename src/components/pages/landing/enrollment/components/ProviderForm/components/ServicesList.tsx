/* eslint-disable import/prefer-default-export */
// External Dependencies
import { useState } from 'react';
import { Select, MenuItem, useMediaQuery } from '@material-ui/core';
import { useFormikContext } from 'formik';
import { theme } from '@care/material-ui-theme';
import { Typography } from '@care/react-component-lib';

// Internal Constants
import { PROVIDER_SERVICES_DESKTOP, PROVIDER_SERVICES_MOBILE } from '../constants';

// Styles
import useStyles from './ServicesList.styles';

function ServicesList() {
  // Styles
  const classes = useStyles();
  const isDesktop = useMediaQuery(theme.breakpoints.up('md'));

  const formik = useFormikContext<{ providerService: string }>();
  const listOfServices = isDesktop ? PROVIDER_SERVICES_DESKTOP : PROVIDER_SERVICES_MOBILE;

  // Component's State
  const [typeOfService, setTypeOfService] = useState('');

  // Handlers
  const onServiceValueChange = (value: string) => {
    formik.setFieldValue('providerService', value);
    setTypeOfService(value as string);
  };

  return (
    <>
      <Select
        name="selectJobType"
        className={
          !typeOfService
            ? `${classes.inputContainer} ${classes.empty}`
            : `${classes.inputContainer}`
        }
        value={typeOfService}
        displayEmpty
        onChange={(e) => {
          onServiceValueChange(e.target.value as string);
        }}>
        <MenuItem value="" disabled>
          Select one:
        </MenuItem>
        {listOfServices.map((service: any) => {
          return (
            <MenuItem value={service.value} key={service.value}>
              {service.label}
            </MenuItem>
          );
        })}
      </Select>
      {formik.errors.providerService && (
        <Typography careVariant="info1" className={classes.error}>
          Please select a job
        </Typography>
      )}
    </>
  );
}

export { ServicesList };
