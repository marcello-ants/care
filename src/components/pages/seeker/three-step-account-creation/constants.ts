import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
