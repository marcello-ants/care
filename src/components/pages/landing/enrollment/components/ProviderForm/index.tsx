// External Dependencies
import { useCallback } from 'react';
import getConfig from 'next/config';
import { FormikProvider } from 'formik';
import Button from '@material-ui/core/Button';
import { Box, Hidden } from '@material-ui/core';
import { Typography } from '@care/react-component-lib';

// Internal Dependencies
import FormikInlineTextField from '@/components/blocks/FormikInlineTextField';
import EmailField from '@/components/features/accountCreation/EmailField';
import PasswordField from '@/components/features/accountCreation/PasswordField';
import { HearAboutUsField } from '@/components/HearAboutUs';
import { useProviderEnrollmentForm } from './utils/formHandler';
import { ServicesList } from './components/ServicesList';
import ProviderError from '../ProviderError';
import ProviderDisclaimer from '../ProviderDisclaimer';

// Styles
import useStyles from './styles';

// Constants
const {
  publicRuntimeConfig: { CZEN_GENERAL_LOGIN },
} = getConfig();

function SeekerForm() {
  // Styles
  const classes = useStyles();

  // Formik
  const { formik, isSubmitError } = useProviderEnrollmentForm();

  // Handlers
  const handleFormSubmission = useCallback(() => {
    if (!formik.isValid || formik.isSubmitting) {
      return;
    }

    formik.handleSubmit();
  }, []);

  return (
    <>
      {isSubmitError && <ProviderError />}
      <FormikProvider value={formik}>
        <Typography variant="h2" className={classes.sectionTitle}>
          <span>What job are you most</span>{' '}
          <Hidden smUp>
            <br />
          </Hidden>{' '}
          <span>interested in?</span>
        </Typography>
        <ServicesList />
        <Box mt={5}>
          <Typography variant="h2" className={`${classes.sectionTitle}`}>
            <span>Create your account</span>
            <Typography careVariant="disclaimer1" className={classes.sectionTitleDisclaimer}>
              *required
            </Typography>
          </Typography>
        </Box>
        <Box className={classes.inputContainer}>
          <EmailField
            id="email"
            name="email"
            label="Email*"
            czenGeneralLoginUrl={CZEN_GENERAL_LOGIN}
            showErrorWhenTouched
          />
        </Box>
        <Box className={classes.inputsContainer}>
          <PasswordField
            id="password"
            name="password"
            label="Password*"
            defaultHelperText=""
            showErrorWhenTouched
          />
        </Box>
        <Box className={classes.inputContainer} mb={2}>
          <FormikInlineTextField id="providerZip" name="providerZip" label="Zip code*" />
        </Box>
        <Box className={classes.inputContainer}>
          <HearAboutUsField name="howDidYouHearAboutUs" />
        </Box>
        <Box mb={4}>
          <ProviderDisclaimer />
        </Box>
        <Button
          color="primary"
          variant="contained"
          size="large"
          className={classes.submitButton}
          disabled={formik.isSubmitting}
          onClick={handleFormSubmission}>
          Join now
        </Button>
      </FormikProvider>
      <Typography variant="body1" className={classes.companyInquiries}>
        <span>For company inquiries</span>{' '}
        <a href="https://www.care.com/business-p1369-q59964639.html">sign up here</a>
        <span>.</span>
      </Typography>
    </>
  );
}

export default SeekerForm;
