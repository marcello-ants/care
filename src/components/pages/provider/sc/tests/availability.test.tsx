import React from 'react';
import { useRouter } from 'next/router';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { cloneDeep } from 'lodash-es';
import DayJSUtils from '@date-io/dayjs';
import { MockedProvider } from '@apollo/client/testing';
import Availability from '@/pages/provider/sc/availability';
import { AppStateProvider } from '@/components/AppState';
import { SENIOR_CARE_PROVIDER_AVAILABILITY_UPDATE } from '@/components/request/GQL';
import { initialAppState } from '@/state';
import { generateProviderAvailabilityUpdateInput } from '@/utilities/gqlPayloadHelper';
import { SeniorCareProviderAvailabilityUpdateInput } from '@/__generated__/globalTypes';
import { PROVIDER_ROUTES } from '@/constants';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

const initialStateClone = cloneDeep(initialAppState);

const initialState = {
  ...initialStateClone,
  provider: {
    ...initialStateClone.provider,
    recurring: {
      ...initialStateClone.provider.recurring,
      // use a date far into the future to avoid having to update snapshots for awhile
      start: '12-31-2100',
    },
  },
};

const successMock = {
  request: {
    query: SENIOR_CARE_PROVIDER_AVAILABILITY_UPDATE,
    variables: {
      input: {
        ...generateProviderAvailabilityUpdateInput(initialState.provider),
        schedule: {
          sunday: {
            blocks: [{ start: '06:00', end: '12:00' }],
          },
        },
      } as SeniorCareProviderAvailabilityUpdateInput,
    },
  },
  result: {
    data: {
      seniorCareProviderAvailabilityUpdate: {
        __typename: 'SeniorCareProviderAvailabilityUpdateSuccess',
      },
    },
  },
};

const errorMock = {
  request: {
    query: SENIOR_CARE_PROVIDER_AVAILABILITY_UPDATE,
    variables: {
      input: {
        ...generateProviderAvailabilityUpdateInput(initialState.provider),
        schedule: {
          monday: {
            blocks: [{ start: '06:00', end: '12:00' }],
          },
        },
      } as SeniorCareProviderAvailabilityUpdateInput,
    },
  },
  error: new Error('An error occurred'),
};

let mockRouter: any | null = null;

function renderPage() {
  return render(
    <MockedProvider mocks={[errorMock, successMock]} addTypename={false}>
      <MuiPickersUtilsProvider utils={DayJSUtils}>
        <AppStateProvider initialStateOverride={initialState}>
          <Availability />
        </AppStateProvider>
      </MuiPickersUtilsProvider>
    </MockedProvider>
  );
}

beforeEach(() => {
  mockRouter = {
    push: jest.fn(),
    pathname: '',
    asPath: '',
  };
  (useRouter as jest.Mock).mockReturnValue(mockRouter);
});

afterEach(() => {
  // cleanup on exiting
  mockRouter = null;
});

describe('Availability component', () => {
  it('matches snapshot', () => {
    const { asFragment } = renderPage();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should navigate to the headline-bio page after successfully saving the selections', async () => {
    renderPage();
    fireEvent.click(screen.getByText('Sun'));
    fireEvent.click(screen.getByText('Mornings'));
    fireEvent.click(screen.getByRole('button', { name: 'Next' }));
    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_ROUTES.HEADLINE_BIO));
  });

  it('should display a message when an error is thrown', async () => {
    renderPage();
    fireEvent.click(screen.getByText('Mon'));
    fireEvent.click(screen.getByText('Mornings'));
    fireEvent.click(screen.getByRole('button', { name: 'Next' }));
    await waitFor(() =>
      expect(screen.getByText(/We weren't able to save your availability/)).toBeVisible()
    );
  });

  it('should  navigate to the headline-bio page after acknowledging the error message', async () => {
    renderPage();
    fireEvent.click(screen.getByText('Mon'));
    fireEvent.click(screen.getByText('Mornings'));
    fireEvent.click(screen.getByRole('button', { name: 'Next' }));
    await screen.findByText(/We weren't able to save your availability/);
    fireEvent.click(screen.getByRole('button', { name: 'Got it' }));
    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_ROUTES.HEADLINE_BIO));
  });
});
