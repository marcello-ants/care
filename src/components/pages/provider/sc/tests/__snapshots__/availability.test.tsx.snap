// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Availability component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-header-13 MuiTypography-h2"
      >
        When are you available to work?
      </h2>
    </div>
    <div
      class="MuiGrid-root makeStyles-datePickerContainer-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root makeStyles-dateSubheader-6 MuiTypography-h4"
      >
        Dates
      </h4>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-root-28 makeStyles-is-filled-31"
        data-testid="keyboardDatePicker"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-error Mui-error MuiFormLabel-filled"
          data-shrink="true"
          for="startDate"
          id="startDate-label"
        >
          Estimated start date
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        >
          <input
            aria-describedby="startDate-helper-text"
            aria-invalid="true"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            id="startDate"
            type="text"
            value="12/31/2100"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-root-34"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.79 4.49h-.47v-.94a.63.63 0 10-1.25 0v1H7.85v-1a.67.67 0 00-.62-.63.67.67 0 00-.63.63v1h-.47A3.13 3.13 0 003 7.7V18a3.14 3.14 0 003.13 3.13h11.74a2.83 2.83 0 002.19-.94A3.13 3.13 0 0021 18V7.62a3.27 3.27 0 00-3.21-3.13zM6.13 5.82h.47v.7a.68.68 0 00.63.63.68.68 0 00.62-.63v-.7h8.22v.7a.63.63 0 101.25 0v-.7h.47a1.86 1.86 0 011.8 1.72H4.25a1.93 1.93 0 011.88-1.72zm13 13.3a1.85 1.85 0 01-1.25.55H6.13a1.77 1.77 0 01-1.8-1.8V8.79h15.34v9.08a1.85 1.85 0 01-.55 1.25z"
                    fill="#182c42"
                  />
                  <path
                    d="M7.77 13.17a.86.86 0 000-1.72.86.86 0 000 1.72zM12 13.17a.86.86 0 10-.86-.86.85.85 0 00.86.86zM16.15 13.17a.86.86 0 10-.86-.86.86.86 0 00.86.86zM7.77 17.09a.86.86 0 10-.86-.86.86.86 0 00.86.86zM12 17.09a.86.86 0 10-.86-.86.86.86 0 00.86.86zM16.15 17.09a.86.86 0 000-1.72.86.86 0 000 1.72z"
                    fill="#182c42"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <p
          class="MuiFormHelperText-root Mui-error MuiFormHelperText-filled"
          id="startDate-helper-text"
        >
          Date should not be after maximal date
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-datePickerContainer-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-root-28"
        data-testid="keyboardDatePicker"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="endDate"
          id="endDate-label"
        >
          Estimated end date
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            id="endDate"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-root-35"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.79 4.49h-.47v-.94a.63.63 0 10-1.25 0v1H7.85v-1a.67.67 0 00-.62-.63.67.67 0 00-.63.63v1h-.47A3.13 3.13 0 003 7.7V18a3.14 3.14 0 003.13 3.13h11.74a2.83 2.83 0 002.19-.94A3.13 3.13 0 0021 18V7.62a3.27 3.27 0 00-3.21-3.13zM6.13 5.82h.47v.7a.68.68 0 00.63.63.68.68 0 00.62-.63v-.7h8.22v.7a.63.63 0 101.25 0v-.7h.47a1.86 1.86 0 011.8 1.72H4.25a1.93 1.93 0 011.88-1.72zm13 13.3a1.85 1.85 0 01-1.25.55H6.13a1.77 1.77 0 01-1.8-1.8V8.79h15.34v9.08a1.85 1.85 0 01-.55 1.25z"
                    fill="#182c42"
                  />
                  <path
                    d="M7.77 13.17a.86.86 0 000-1.72.86.86 0 000 1.72zM12 13.17a.86.86 0 10-.86-.86.85.85 0 00.86.86zM16.15 13.17a.86.86 0 10-.86-.86.86.86 0 00.86.86zM7.77 17.09a.86.86 0 10-.86-.86.86.86 0 00.86.86zM12 17.09a.86.86 0 10-.86-.86.86.86 0 00.86.86zM16.15 17.09a.86.86 0 000-1.72.86.86 0 000 1.72z"
                    fill="#182c42"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root makeStyles-daySubheader-7 MuiTypography-h4"
      >
        Days
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <ul
        class="MuiList-root makeStyles-careDays-11 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-root-37 MuiListItem-gutters"
        >
          <input
            id="selector.careDays.0"
            name="careDays"
            readonly=""
            type="checkbox"
            value="sunday"
          />
          <span
            class="control-indicator"
          >
            <span
              class="makeStyles-indicator-38"
            >
              S
            </span>
          </span>
          <span
            class="control-label"
          >
            Sun
          </span>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-37 MuiListItem-gutters"
        >
          <input
            id="selector.careDays.1"
            name="careDays"
            readonly=""
            type="checkbox"
            value="monday"
          />
          <span
            class="control-indicator"
          >
            <span
              class="makeStyles-indicator-38"
            >
              M
            </span>
          </span>
          <span
            class="control-label"
          >
            Mon
          </span>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-37 MuiListItem-gutters"
        >
          <input
            id="selector.careDays.2"
            name="careDays"
            readonly=""
            type="checkbox"
            value="tuesday"
          />
          <span
            class="control-indicator"
          >
            <span
              class="makeStyles-indicator-38"
            >
              T
            </span>
          </span>
          <span
            class="control-label"
          >
            Tue
          </span>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-37 MuiListItem-gutters"
        >
          <input
            id="selector.careDays.3"
            name="careDays"
            readonly=""
            type="checkbox"
            value="wednesday"
          />
          <span
            class="control-indicator"
          >
            <span
              class="makeStyles-indicator-38"
            >
              W
            </span>
          </span>
          <span
            class="control-label"
          >
            Wed
          </span>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-37 MuiListItem-gutters"
        >
          <input
            id="selector.careDays.4"
            name="careDays"
            readonly=""
            type="checkbox"
            value="thursday"
          />
          <span
            class="control-indicator"
          >
            <span
              class="makeStyles-indicator-38"
            >
              T
            </span>
          </span>
          <span
            class="control-label"
          >
            Thu
          </span>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-37 MuiListItem-gutters"
        >
          <input
            id="selector.careDays.5"
            name="careDays"
            readonly=""
            type="checkbox"
            value="friday"
          />
          <span
            class="control-indicator"
          >
            <span
              class="makeStyles-indicator-38"
            >
              F
            </span>
          </span>
          <span
            class="control-label"
          >
            Fri
          </span>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-37 MuiListItem-gutters"
        >
          <input
            id="selector.careDays.6"
            name="careDays"
            readonly=""
            type="checkbox"
            value="saturday"
          />
          <span
            class="control-indicator"
          >
            <span
              class="makeStyles-indicator-38"
            >
              S
            </span>
          </span>
          <span
            class="control-label"
          >
            Sat
          </span>
        </li>
      </ul>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root makeStyles-timeSubheader-8 MuiTypography-h4"
      >
        Times
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <ul
        class="MuiList-root makeStyles-rootInline-36 makeStyles-careTimes-12 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-root-39 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-42"
            id="selector.careTimes.0"
            name="careTimes"
            readonly=""
            type="checkbox"
            value="morning"
          />
          <div
            class="MuiBox-root MuiBox-root-50 makeStyles-controlIndicator-43 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Mornings
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-39 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-42"
            id="selector.careTimes.1"
            name="careTimes"
            readonly=""
            type="checkbox"
            value="afternoon"
          />
          <div
            class="MuiBox-root MuiBox-root-51 makeStyles-controlIndicator-43 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Afternoons
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-39 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-42"
            id="selector.careTimes.2"
            name="careTimes"
            readonly=""
            type="checkbox"
            value="evening"
          />
          <div
            class="MuiBox-root MuiBox-root-52 makeStyles-controlIndicator-43 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Evenings
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-39 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-42"
            id="selector.careTimes.3"
            name="careTimes"
            readonly=""
            type="checkbox"
            value="overnight"
          />
          <div
            class="MuiBox-root MuiBox-root-53 makeStyles-controlIndicator-43 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Overnight
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <button
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-specificTimesLink-5 MuiLink-button MuiTypography-body2 MuiTypography-colorPrimary"
      >
        Add specific times instead
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h6
        class="MuiTypography-root makeStyles-adjustLaterText-9 MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Don't worry, you can always adjust these times later.
      </h6>
    </div>
    <div
      class="makeStyles-wrapper-54"
    >
      <div
        class="makeStyles-fade-55"
        style="position: relative; width: auto; margin-bottom: -24px; padding: 32px 40px 24px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled MuiButton-fullWidth Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
